src/bin/interactive.rs

```diff
@@ -231,54 +234,59 @@ fn menu_op_offline(addr: Address) -> AddrOp {
     match cmd {
         CMI::Export => AddrOp::Run(Box::new(Command::Export {
             addr: Some(addr),
-            dir: prompt::request_dir("export keys"),
+            dir: prompt::request_dir("export keys", settings.profile.clone()),
         })),
-        CMI::Return => AddrOp::Return,
+        CMI::Back => AddrOp::Back,
         _ => unreachable!(),
     }
 }
 
 /// Allows the user to load a wallet interactively
 pub(crate) fn load_wallet(
     wallet_path: &WalletPath,
+    settings: &Settings,
 ) -> Result<Wallet<WalletFile>, Error> {
     // find wallets in the specified data directory
-    let wallet_dir = wallet_path.dir().unwrap_or_else(WalletPath::default_dir);
-    let wallets_found = WalletPath::wallets_in(&wallet_dir)?;
+    let wallet_dir = wallet_path
+        .dir()
+        .unwrap_or_else(|| settings.profile.clone())
+        .join("wallet.dat");
+
+    let wallet_found = wallet_dir.exists().then(|| wallet_dir.clone());
```

Comment:
As I tried to explain during the call, if you don't clone it you're moving `wallet_dir`, since here the code creates an `Option<PathBuf>` if you're moving, then you cannot have a reference to `wallet_dir` anymore afterwards.
We might avoid cloning only if we change how the menu and the `WalletPath` works, but IMVHO is not worthy.